File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        count = 0
        for i in range(n-2):
            if nums[i] == 0:
                count += 1
                self.flip(nums,i)
                self.flip(nums,i+1)
                self.flip(nums,i+2)
        valid = True
        for i in range(n):
            if nums[i] == 0:
                valid = False
                break
        if valid:
            return count
        else:
            return -1

    def flip(self, nums:List[int],pos) -> None:
        if nums[pos] == 1:
            nums[pos] = 0
        else:
            nums[pos] = 1
                
        

File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.java
class Solution {
    public int minOperations(int[] nums) {
        int flipcount = 0;
        for(int i = 0; i <= nums.length-3; i++){
            if(nums[i]==0){
                //flip curr and next two nums
                flipcount++;
                flip(nums,i);
                flip(nums,i+1);
                flip(nums,i+2);
            }
        }
        boolean valid = true;
        for(int num : nums){
            if(num!=1){
                valid = false;
                break;
            }
        }
        if(valid){
            return flipcount;
        }
        else
            return -1;
    }
    public void flip(int[]nums, int pos){
        if(nums[pos] == 1)
            nums[pos] = 0;
        else
            nums[pos] = 1;
        return;
    }
}